packages/teams-js/src/public/authentication.ts

```diff
@@ -28,39 +33,53 @@ export namespace authentication {
+   *
+   * You can learn more details about this flow [here](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab)
+   *
+   * This function is *not* needed for "standard" Azure SSO usage. Using {@link getAuthToken} is usually sufficient in that case (learn more
+   * [here](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview))
    *
    * @remarks
    * The authentication flow must start and end from the same domain, otherwise success and failure messages won't be returned to the window that initiated the call.
+   * As described on the [3rd party OAuth provider documentation](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab), the
+   * authentication flow starts and ends at an endpoint on your own service (with a redirect round-trip to the 3rd party identity provider in the middle).
    *
-   * @param authenticateParameters - The parameters for the authentication request. It is a required parameter since v2 upgrade
+   * @param authenticateParameters - Parameters describing the authentication window used for executing the authentication flow
    *
-   * @returns Promise that will be fulfilled with the result from the authentication pop-up if successful.
+   * @returns `Promise` that will be fulfilled with the result from the authentication pop-up, if successful. The string in this result is provided in the parameter
+   * passed by your app when it calls {@link notifySuccess} in the pop-up window after returning from the identity provider redirect.
    *
-   * @throws Error if the authentication request fails or is canceled by the user.
+   * @throws `Error` if the authentication request fails or is canceled by the user. This error is provided in the parameter passed by your app when it calls
+   * {@link notifyFailure} in the pop-up window after returning from the identity provider redirect. However, in some cases it can also be provided by
+   * the infrastructure depending on the failure (e.g., a user cancelation)
    *
    */
   export function authenticate(authenticateParameters: AuthenticatePopUpParameters): Promise<string>;
   /**
    * @deprecated
    * As of 2.0.0, please use {@link authentication.authenticate authentication.authenticate(authenticateParameters: AuthenticatePopUpParameters): Promise\<string\>} instead.
    *
-   * Initiates an authentication request, which opens a new window with the specified settings.
-   *
-   * @remarks
-   * The authentication flow must start and end from the same domain, otherwise success and failure messages won't be returned to the window that initiated the call.
+   * The documentation for {@link authentication.authenticate authentication.authenticate(authenticateParameters: AuthenticatePopUpParameters): Promise\<string\>} applies
+   * to this function. Please see that documentation.
```

Comment:
Can probably omit "Please see that documentation." sentence, as its already implied by the overall comment